### PR TITLE
Support development builds in pull-requests

### PR DESCRIPTION
**Pull request by agners**
This adds support for development builds in pull-requests. This allows to test a change before merging on actual hardware.

A PR needs to be labeled with `run-dev-build` as well as the board labels the build should be done for.

The development build get a custom version number with the PR number prepended (e.g. `7.0.dev2021071400123`).

### CODE REVIEW

File: .github/workflows/dev.yml [hunk 1]
```diff
@@ -4,15 +4,18 @@ name: Development build
 
 on:
   workflow_dispatch:
+  pull_request:
```
Comment (ludeeus) [inline .github/workflows/dev.yml:7]: This will fail when the pull request is opened from a fork

Comment (agners) [reply .github/workflows/dev.yml:7]: This is from a fork... Seems to work?

Comment (ludeeus) [reply .github/workflows/dev.yml:7]: Forks does not have access to secrets

Comment (agners) [reply .github/workflows/dev.yml:7]: I see, but using `pull_request_target` should work right? I just can't test it until merged :)

Comment (agners) [reply .github/workflows/dev.yml:7]:
```suggestion
  pull_request_target:
````

Comment (ludeeus) [reply .github/workflows/dev.yml:7]: it does, use with care <https://securitylab.github.com/research/github-actions-preventing-pwn-requests/>

**Pull request merged**